### ISSUE TITLE
'ASTTemplateParamNonType' object has no attribute 'isPack'

### ISSUE THREAD

**Issue by bluescarni** [sphinx-doc/sphinx#5126]
Since Sphinx 1.7, I cannot build anymore the documentation of my C++ projects. The errors all look like this:

```console
Exception occurred:
  File "/usr/lib64/python3.6/site-packages/sphinx/domains/cpp.py", line 3529, in isSpecialization
    if param.isPack != isArgPackExpansion:
AttributeError: 'ASTTemplateParamNonType' object has no attribute 'isPack'
```

The errors seem to be triggered when concepts are involved. For instance, in one specific case I have a concept defined like this:

```rst
.. cpp:concept:: template <typename T> piranha::CppIntegral
```

which is then used in a class template specialization like this:

```rst
.. cpp:class:: template <piranha::CppIntegral T, piranha::CppIntegral U> piranha::binomial_impl<T, U>
```

Removing the documentation of the class specialization gets rid of the error. Sphinx 1.6.x works fine and produces the expected output.

Note that I am using directly Sphinx's C++ domain, I am **not** going through the Breathe+Doxygen route.

**Comment by jakobandersen** [comment]
Thanks for reporting. It should be fixed in the upcoming version 1.8.

**Comment by bluescarni** [comment]
@jakobandersen Thank you! I will try it out on the beta.
